### PR TITLE
adds federated service architecture and explains working

### PR DESCRIPTION
**Pull request by JBAhire**
This PR adds
- federated service architecture wrt Hypertrace query architecture.
- working of federated service
- build steps and image source

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,9 +3,25 @@
...
-- **Attribute service:** Service to provide CRUD operations for the attributes in Hypertrace.
-- **Entity service:** Service that provides CRUD operations for differently identified entities of observed applications.
-- **Gateway service:** An entry service that acts as a single access point for querying data from other services like entity-service, query-service, etc
-- **Query service:** Query service exposes APIs to query data from the underlying stores. 
-- **Hypetrace UI:** Serves static UI resources 
+Hypertrace federated service combines [Hypertrace-UI](https://github.com/hypertrace/hypertrace-ui), [Hypertrace-GraphQL](https://github.com/hypertrace/hypertrace-graphql), [Gateway service](https://github.com/hypertrace/gateway-service), [Attribute service](https://github.com/hypertrace/attribute-service), [Query service](https://github.com/hypertrace/query-service), [Entity service](https://github.com/hypertrace/entity-service) into single service. This is used in hypertrace standalone deployment to make the deployment compact and use less resource.
```
Comment (jcchavezs) [inline README.md:6]:
```suggestion
Hypertrace Federated Service combines [Hypertrace-UI](https://github.com/hypertrace/hypertrace-ui), [Hypertrace-GraphQL](https://github.com/hypertrace/hypertrace-graphql), [Gateway service](https://github.com/hypertrace/gateway-service), [Attribute service](https://github.com/hypertrace/attribute-service), [Query service](https://github.com/hypertrace/query-service), [Entity service](https://github.com/hypertrace/entity-service) into single service. This is used in hypertrace standalone deployment to make the deployment compact and use less resource.
```

File: README.md [hunk 2]
```diff
@@ -3,9 +3,25 @@
...
+
+## How Federated service works?
```
Comment (jcchavezs) [inline README.md:8]:
```suggestion
## How Hypertrace Federated Service works?
```

File: README.md [hunk 3]
```diff
@@ -3,9 +3,25 @@
...
+
+| ![space-1.jpg](https://hypertrace-docs.s3.amazonaws.com/federated-service.png) | 
+|:--:| 
+| *Hypertrace Query Architecture* |
+
+In Hypertrace Federated service,
```
Comment (jcchavezs) [inline README.md:14]:
```suggestion
In Hypertrace Federated Service, 
```

**Comment by jcchavezs** [comment]
It looks good to me, tho I see some inconsistencies between uppercase/lowercase usage when it comes to service names. Maybe it is a good idea that a native reviews it? cc @adriancole

File: README.md (hunk 1 above)
Comment (adriancole) [reply README.md]: yeah I think we shouldn't bother with changing query-service etc to "Query Service" as it makes more work when renaming and less concrete to container names.

File: README.md [hunk 4]
```diff
@@ -3,9 +3,25 @@
...
+
+## How federated service works?
+
+| ![space-1.jpg](https://hypertrace-docs.s3.amazonaws.com/federated-service.png) | 
+|:--:| 
+| *Hypertrace Query Architecture* |
+
+In Hypertrace federated service, 
+- GraphQL service is being used by UI and attribute-service fetches all relevant attributes to the scope of what is being shown.
+- gateway-service provides single access point which routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by downstream services.
```
Comment (buchi-busireddy) [inline README.md:16]:
```suggestion
- gateway-service provides single access point which routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by upstream services. 
```

**Review by buchi-busireddy — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -3,9 +3,25 @@
...
+
+## How federated service works?
+
+| ![space-1.jpg](https://hypertrace-docs.s3.amazonaws.com/federated-service-arch.png) | 
+|:--:| 
+| *Hypertrace Query Architecture* |
+
+In Hypertrace federated service, 
+- GraphQL service is being used by UI and attribute-service fetches all relevant attributes to the scope of what is being shown.
+- gateway-service provides single access point which routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by upstream services.
```
Comment (kotharironak) [inline README.md:16]:
```suggestion
- The gateway-service provides a single access point that routes queries to corresponding downstream service based on the source of attributes and then does appropriate type conversion of data returned by upstream services. 
```

**Review by kotharironak — DISMISSED** [review]

**Review by kotharironak — APPROVED** [review]

**Pull request merged**